### ISSUE TITLE
nmcli: MODULE FAILURE on Ubuntu 16.04

### ISSUE THREAD

**Issue by maxsolar** [ansible/ansible#19792]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

nmcli

##### ANSIBLE VERSION

```
ansible: 2.2.0.0
nmcli: 1.2.2
libnm-glib-dev: 1.2.2-0ubuntu0.16.04.3
```

##### CONFIGURATION

host_key_checking = False

##### OS / ENVIRONMENT

Ubuntu 16.04 LTS amd64

##### SUMMARY

while using nmcli module to manipulate Ubuntu network-manager, it failed with MODULE FAILURE.

##### STEPS TO REPRODUCE

```yaml
---
- name: smartNetworker.yml
  hosts: local
  user: buildteam
  become: yes
  vars:
    ipv4: '10.7.40.59'
    gateway4: '10.7.40.252'
    dns4_server: [ "10.9.128.16", "10.9.32.1", "10.5.8.2" ]
    interface: 'enp6s0'
    hostname: 'Build-59'
    connection: "smart-{{interface}}"

  tasks:
    - name: <smartNetworker.yml> change nmcli configuration...
      nmcli:
          type: 'ethernet'
          ifname: "{{ interface }}"
          conn_name: "{{ connection }}"
          ip4: "{{ ipv4 }}/21"
          gw4: "{{ gateway4 }}"
          dns4: "{{ dns4_server }}"
          state: present

    - name: <smartNetworker.yml> change hostname
      shell: hostnamectl set-hostname "{{hostname}}"

```

##### EXPECTED RESULTS

All tasks should end with OK.

##### ACTUAL RESULTS

```
"module_stderr": "OpenSSH_7.2p2 Ubuntu-4ubuntu2.1, OpenSSL 1.0.2g  1 Mar 2016\r\ndebug1: Reading configuration data /home/buildteam/.ssh/config\r\ndebug1: /home/buildteam/.ssh/config line 99: Applying options for 127.*.*.*\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 19: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 4324\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\nShared connection to 127.0.0.1 closed.\r\n", 
    "module_stdout": "/tmp/ansible_ZBvbfg/ansible_module_nmcli.py:393: PyGIWarning: NetworkManager was imported without specifying a version first. Use gi.require_version('NetworkManager', '1.0') before import to ensure that the right version gets loaded.\r\n  from gi.repository import NetworkManager, NMClient\r\n/tmp/ansible_ZBvbfg/ansible_module_nmcli.py:393: PyGIWarning: NMClient was imported without specifying a version first. Use gi.require_version('NMClient', '1.0') before import to ensure that the right version gets loaded.\r\n  from gi.repository import NetworkManager, NMClient\r\nTraceback (most recent call last):\r\n  File \"/tmp/ansible_ZBvbfg/ansible_module_nmcli.py\", line 1086, in <module>\r\n    main()\r\n  File \"/tmp/ansible_ZBvbfg/ansible_module_nmcli.py\", line 1060, in main\r\n    if nmcli.connection_exists():\r\n  File \"/tmp/ansible_ZBvbfg/ansible_module_nmcli.py\", line 572, in connection_exists\r\n    connections=self.list_connection_info()\r\n  File \"/tmp/ansible_ZBvbfg/ansible_module_nmcli.py\", line 541, in list_connection_info\r\n    proxy=bus.get_object(service_name, \"/org/freedesktop/NetworkManager/Settings\")\r\n  File \"/usr/lib/python2.7/dist-packages/dbus/bus.py\", line 241, in get_object\r\n    follow_name_owner_changes=follow_name_owner_changes)\r\n  File \"/usr/lib/python2.7/dist-packages/dbus/proxies.py\", line 248, in __init__\r\n    self._named_service = conn.activate_name_owner(bus_name)\r\n  File \"/usr/lib/python2.7/dist-packages/dbus/bus.py\", line 180, in activate_name_owner\r\n    self.start_service_by_name(bus_name)\r\n  File \"/usr/lib/python2.7/dist-packages/dbus/bus.py\", line 278, in start_service_by_name\r\n    'su', (bus_name, flags)))\r\n  File \"/usr/lib/python2.7/dist-packages/dbus/connection.py\", line 651, in call_blocking\r\n    message, timeout)\r\ndbus.exceptions.DBusException: org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.NetworkManager was not provided by any .service files\r\n", 
    "msg": "MODULE FAILURE"
```

**Comment by Loustik** [comment]
Why did you close this issue without any explanation / details?

**Comment by ashishbista** [comment]
I'm getting the same error. Is it ever fixed?

**Comment by maxsolar** [comment]
This issue has been fixed since Ansible 2.3.1.0 while my current installed version is 2.3.2.0 and we have no such issue.
My bad to close this issue without explanations.
